bootstrap/config/config.go

```diff
@@ -1271,3 +1280,76 @@ func GetInsecureSecretDataFullPath(secretName, key string) string {
 	return fmt.Sprintf("%s/%s/%s/%s/%s",
 		writableKey, insecureSecretsKey, secretName, secretDataKey, key)
 }
+
+func PutMissingConfigToConfigProvider(yamlFilePath string, secretProvider interfaces.SecretProvider, configClient configuration.Client, lc logger.LoggingClient) error {
+	keyValues, err := KeyValuesFromYamlFile(yamlFilePath, secretProvider, lc)
+	if err != nil {
+		return fmt.Errorf("failed to put missing config by yaml file %s: %s", yamlFilePath, err.Error())
+	}
+
+	for key, value := range keyValues {
+		if value == nil {
+			continue
+		}
+		hasSubConfig, err := configClient.HasSubConfiguration(key)
+		if err != nil {
+			return fmt.Errorf("failed to check config %s: %s", key, err.Error())
+		}
+		if hasSubConfig {
+			continue
+		}
+		if err = configClient.PutConfigurationValue(key, []byte(fmt.Sprint(value))); err != nil {
+			return fmt.Errorf("failed to push common configuration key %s with value %v: %s", key, value, err.Error())
```

Comment:
From the error log, does the code here only handle common config? How about the private config?